pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -378,6 +377,22 @@ func (pl *VolumeRestrictions) isSchedulableAfterPodDeleted(logger klog.Logger, p
 		return framework.Queue, nil
 	}
 
+	// Check if the deleted pod uses the same PVC with the ReadWriteOncePod access mode.
+	// Get the access mode of the PVCs may be expensive, so we only compare the PVC names.
+	// TODO: If we observe good throughput, we will compare the access mode of the PVCs.
+	// https://github.com/kubernetes/kubernetes/issues/124566
```

Comment:
```suggestion
	// Return Queue if a deleted pod uses the same PVC since the pod may be unschedulable due to the ReadWriteOncePod access mode of the PVC.
	//
	// For now, we don't actually fetch PVC and check the access mode because that operation could be expensive.
	// Once the observability around QHint is established,
	// we may want to do that depending on how much the operation would impact the QHint latency negatively.
	// https://github.com/kubernetes/kubernetes/issues/124566
```